src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,140 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() { // Constructor
+        elementData = new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacityInternal(size + 1);
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        ensureCapacityInternal(size + 1);
+        System.arraycopy(elementData, index, elementData, index + 1,size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        int oldElementDataSize = size;
+        ensureCapacityInternal(size + list.size());
+        for (int i = 0; i < list.size(); i++) {
+            elementData[oldElementDataSize++] = list.get(i);
+        }
+        size = size + list.size();
+    }
+
+    //////// Methods for addition
+
+    private void ensureCapacityInternal(int minCapacity) {
+        if (minCapacity - elementData.length > 0) {
```

Comment:
```suggestion
        if (minCapacity > elementData.length) {
```

Reply:
Fixed. Thank you)   this is embarrassing)))